cli/studio.py

```diff
@@ -185,7 +190,60 @@ def replay(directory: Sequence[str], timestep: float, endpoint: str):
             )
 
 
+@scenario_cli.command(
+    name="preview", help="Build and run the scenario for viewing in Envision."
+)
+@click.option(
+    "--envision",
+    is_flag=True,
+    default=True,
+    help="Start up Envision server at the specified port when running an experiment",
+)
+@click.option(
+    "-p",
+    "--envision_port",
+    help="Port on which Envision will run.",
+    default=None,
+)
+@click.argument("scenario", type=click.Path(exists=True), metavar="<scenario>")
+def preview_scenarios(envision, envision_port, scenario):
+    from cli.run import kill_process_group_afterwards
+    from smarts.env.run_env import main
+    from smarts.sstudio.build_scenario import clean_scenario
+    if os.path.exists(os.path.join(scenario, "scenario.py")):
+        subprocess.call(["scl", "scenario", "build", "--clean", f"{scenario}"])
+    with kill_process_group_afterwards():
+        if envision:
+            if envision_port is None:
+                envision_port = 8081
+            subprocess.Popen(
+                [
+                    "scl",
+                    "envision",
+                    "start",
+                    "-s",
+                    "./scenarios",
+                    "-p",
+                    str(envision_port),
+                ],
+            )
+            # Just in case: give Envision a bit of time to warm up
+            time.sleep(2)
+            url = "http://localhost:" + str(envision_port)
+            webbrowser.open_new_tab(url)
+
+        if (not envision) and envision_port:
+            click.echo(
+                "Port passed without starting up the envision server. Use the --envision option to start the server along with the --envision port option."
+            )
```

Comment:
I have a thought that this might be overthinking that having these two together is an issue. Assuming envision refreshes properly (in the future) it should be entirely reasonable to have this particular case.